handlers/handlers.go

```diff
@@ -28,7 +28,7 @@ func read(w http.ResponseWriter, req *http.Request, cfg *config.Config, rend Ren
 
 	urlQuery := req.URL.Query()
 
-	validatedQueryParams, err := data.ReviewQuery(ctx, cfg, urlQuery)
+	validatedQueryParams, validationProblem, err := data.ReviewQuery(ctx, cfg, urlQuery)
```

Comment:
So instead of handling boolean field, check the error and then make the decision on what to do; something like:

```
        validatedQueryParams, err := data.ReviewQuery(ctx, cfg, urlQuery)
	if err != nil && err != apperrors.<your new error <- needs to be added to apperrors package> {
		log.Error(ctx, "unable to review query", err)
		setStatusCode(w, req, err)
		return
	}
```

Then where you check validateProblem below, change that to `if err == apperrors.<your new error <- needs to be added to apperrors package> {`

See next comment